examples/gan/_gan_mnist_pl/model_def.py

```diff
@@ -0,0 +1,30 @@
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext, DataLoader as DetDataLoader
+from determined.pytorch._lightning import PTLAdapter
+import pl
+
+class GanTrial(PTLAdapter):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        self.dm = pl.MNISTDataModule(batch_size=context.get_global_batch_size())
+        channels, width, height = self.dm.size()
+        lm = pl.GAN(channels, width, height,
+                    batch_size=context.get_global_batch_size(),
+                    lr=context.get_hparam('lr'),
+                    b1=context.get_hparam('b1'),
+                    b2=context.get_hparam('b2'),
+        )
+
+        context.wrap_model(lm.generator.model)
```

Comment:
shouldn't this be called within `PTLAdapter.__init__`?

Reply:
I was thinking we leave this out for the user to call for multi model/optimizer algorithms like GANs

Reply:
You could actually just call wrap_model on the LightningModule

Reply:
wouldn't wrap model need to be called once per nn module?

Reply:
yeah all the submodules will be called on automatically. LightningModule itself is a nn.Module.

Reply:
I moved this into `__PLAdapter.__init__`  we wrap the passed in lightning module for the user and relying on the pytorch's behavior on sub model for nn.Module. If the user has separate models defined that are not part of the lM they'll have to call wrap_model on it themselves in `model_def`. does that sound right?

Reply:
> If the user has separate models defined that are not part of the lM they'll have to call wrap_model on it themselves in model_def. does that sound right?

Yes if they don't define the models to be a member of the lightning module.